ionic_user_interface/src/common/api/route/getRoutesByUser.ts

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios';
+import { routeBaseUrl } from './config';
+import Providers from '@/providers';
+import cacheManager from 'cache-manager';
+
+const memoryCache = cacheManager.caching({
+  store: 'memory',
+  max: 10, // Number of items in cache
+  ttl: 1, // Seconds
+});
+
+const getRoutesByUserUrl = routeBaseUrl + '/route/user';
+
+const getRoutesByUser = async (username: string): Promise<ResponseData> => {
+  const headers = Providers.getLoggedIn().value
+    ? { Authorization: `Bearer ${Providers.getAccessToken().value}` }
+    : {};
+  const response = await axios.get(getRoutesByUserUrl, {
+    headers,
+    params: { username },
+  });
+  return response.data as ResponseData;
+};
+
+// Caches in-memory, disappears on page refresh
+const getRoutesByUserCached = async (username: string): Promise<ResponseData> => {
+  return memoryCache.wrap(username, function () {
+    return getRoutesByUser(username);
+  });
+};
+
+interface ResponseData {
+  Message: string;
+  Items: UserRoute[];
+}
+
+interface UserRoute {
```

Comment:
Actually, this looks to be a superset of GymRoutes, maybe we can subclass? Up to you tho